src/simsopt/geo/surface.py

```diff
@@ -508,6 +572,23 @@ def dmajor_radius_by_dcoeff(self):
         dR_major_ds = (-self.volume() * dmean_area_ds + self.dvolume_by_dcoeff() * mean_area) / mean_area**2
         return dR_major_ds * np.sign(self.volume()) / (2. * np.pi)
 
+    def d2major_radius_by_dcoeff_dcoeff(self):
+        """
+        Return the second derivative of the major radius wrt surface coefficients
+        """
+        V = self.volume()
+        dV_ds = self.dvolume_by_dcoeff()[:, None]
+        dV_dt = self.dvolume_by_dcoeff()[None, :]
+        d2V_dsdt = self.d2volume_by_dcoeffdcoeff()
+        r = self.minor_radius()
+        dr_ds = self.dminor_radius_by_dcoeff()[:, None]
+        dr_dt = self.dminor_radius_by_dcoeff()[None, :]
+        d2r_dsdt = self.d2minor_radius_by_dcoeff_dcoeff()
+
+        return ((6*V*dr_dt*dr_ds)/r**4 - (2*dV_dt*dr_ds)/r**3 - \
+                (2*dr_dt*dV_ds)/r**3 - (2*V*d2r_dsdt)/r**3 + \
+                d2V_dsdt/r**2) * np.sign(V)/(2*np.pi**2)
```

Comment:
```suggestion
        return (
            (6*V*dr_dt*dr_ds)/r**4 - (2*dV_dt*dr_ds)/r**3 - 
            (2*dr_dt*dV_ds)/r**3 - (2*V*d2r_dsdt)/r**3 + 
             d2V_dsdt/r**2
             ) * np.sign(V)/(2*np.pi**2)
```

I thought you were missing a parenthesis :-) but it was correct. in this way is a little bit easier to read